### PR TITLE
Refactor for newer mysql-shell snap

### PR DESCRIPTION
**Pull request by paulomach**
# Issue

~~TLS support is blocked due non updated mysql-shell snap~~ - related to [DPE-722](https://warthogs.atlassian.net/browse/DPE-722)

**EDIT** mysql-shell snap is updated on edge channel.

This PR was repurposed to fix integration tests after TLS refactoring

# Solution

~~Alternate method to install mysql-shell from upstream sources.~~
Using snap from edge channel.
Opportunistic addressed some issues on CI testing.

# Context

Test PR to validate integration testing in CI

### CODE REVIEW

File: tests/integration/test_tls.py [hunk 1]
```diff
@@ -158,6 +159,9 @@ async def test_rotate_tls_key(ops_test: OpsTest) -> None:
     # set key using auto-generated key for each unit
     for unit in ops_test.model.applications[app].units:
         action = await unit.run_action(action_name="set-tls-private-key")
+        # action.wait not always await action status to change
+        # adding some time to address finicky CI test
+        sleep(4)
```
Comment (shayancanonical) [inline tests/integration/test_tls.py:164]: is it possible to use retries instead of sleep here? im always wary of sleeping in tests

**Review by shayancanonical — APPROVED** [review]
lgtm! just one minor nit in the tests

File: tests/integration/test_tls.py (hunk 1 above)
Comment (paulomach) [reply tests/integration/test_tls.py:164]: sure, I think I prefer that also

**Review by WRFitch — APPROVED** [review]

File: tests/integration/test_tls.py (hunk 1 above)
Comment (paulomach) [reply tests/integration/test_tls.py]: 8e9d84b

**Review by shayancanonical — APPROVED** [review]

**Pull request merged**